### ISSUE TITLE
ValueError: setting an array element with a sequence.

### ISSUE THREAD

**Issue by roelant001** [Kamnitsask/deepmedic#11]
While trying to train on my own data, I'm encountering some issues. I was able to trace back most, but am now stuck with the error below. Any suggestions on what might be the cause? Some sub-epochs run fine, while others crash. I'm still trying to find out if there is a correlation with specific scans/patients..

```
Images for subject loaded.
From current subject #19 extracting that many Segments centered on : Foreground=25 Background:25
End of the function that extracts segments. Segments extracted centered on: Foreground=25 Background=25
TIMING: Extracting all the Segments for next Training took time: 179.138042(s)
:=:=:=:=:=:=:=:=: Finished extracting Segments from the images for next Training. :=:=:=:=:=:=:=:=:
Loading Training data for subepoch #23 on shared variable...
Traceback (most recent call last):
  File "deepMedicRun", line 102, in <module>
    deepMedicTrain.deepMedicTrainMain(absPathToTrainConf, absPathToCnnModelGivenInCmdLine, cnnInstanceLoaded, filenameAndPathWhereNewlyCreatedModelWasSaved)
  File "/theano/deepmedic/deepmedic/frontEndModules/deepMedicTrain.py", line 440, in deepMedicTrainMain
    do_training(*trainSessionParameters.getTupleForCnnTraining())
  File "/theano/deepmedic/deepmedic/trainValidateTestVisualiseParallel.py", line 1311, in do_training
    arrayLesionLabelsForPATCHESOfTheImagePartsInGpUForSubepochTraining = np.asarray(lesionLabelsForPATCHESOfTheImagePartsInGpUForSubepochTraining, dtype='float32')
  File "/usr/lib/python2.7/dist-packages/numpy/core/numeric.py", line 460, in asarray
    return array(a, dtype, copy=False, order=order)
ValueError: setting an array element with a sequence.

```

_p.s. Thanks a lot for sharing this tool, first results look very promising!_

**Comment by Kamnitsask** [comment]
Hi Roelant,

Sorry for this. Seems like I should place some checks in these functions to start giving users some appropriate error messages. The functions are improved in some piece of code that I have somewhere, but merging will take some time as I need to do it controllably :-/

Could you please email me the trainingSession.txt log file? It will help me debug this. Please email it at konstantinos.kamnitsas12@imperial.ac.uk.

In the mean time, please make sure that the ground-truth-labels for the subjects that break it do not have different dimensions (I do not remember for sure how it would behave if eg the nii for labels is 4-dimensional, with the 4th dimension empty. Which happens quite often in some people's niis.) and that the ground truth labels have values: 0,1,2,3...,Nclasses, without any values exceeding the Nclasses.

I suspect it could be the first case, with a 4-dimensional nii.

Thanks a lot for the feedback, keep me updated!
Kostas

**Comment by roelant001** [comment]
Hi Kostas,

Thanks for your quick reply and suggestions, I'm going to look into your suggestions for the scans I've already excluded because of issues during loading.
After spending another evening hunting "the scan/segmentation to blame" and experiencing no crashes while training on small chunks of the dataset, I decided to try and run training on the entire dataset(~300 cases at this time) again. It's currently starting epoch 3 without any issues.. I'm not sure what has changed, but it seems to be working (for) now! Maybe it just ran out of memory or something like that. It's nice to have a multi-threaded implementation, but it can make debugging a bit more challenging.

Sorry for having bothered you with my frustrations. If it crashes again I'll sent you the log and config files.

Kind regards, keep up the good work!
Roelant
